### PR TITLE
Check drawIndirectCount enabled by feature and extension

### PR DESCRIPTION
**Pull request by ncesario-lunarg**
Mainly creating this PR to test proposed changes to #2605 while including the tests from #2608 and some additional tests 1.2 tests.

### CODE REVIEW

File: tests/vkpositivelayertests.cpp [hunk 1]
```diff
@@ -3482,6 +3482,122 @@ TEST_F(VkPositiveLayerTest, DrawIndirectCountWithoutFeature) {
...
+    vk::CmdDrawIndexedIndirectCount(m_commandBuffer->handle(), indexed_indirect_buffer.handle(), 0, count_buffer.handle(), 0, 1,
+                                     sizeof(VkDrawIndexedIndirectCommand));
+    m_commandBuffer->EndRenderPass();
+    m_commandBuffer->end();
+
+    m_errorMonitor->VerifyNotFound();
+}
+
+TEST_F(VkPositiveLayerTest, DrawIndirectCountWithFeature) {
+    TEST_DESCRIPTION("Use VK_KHR_draw_indirect_count in 1.2 using the extension");
```
Comment (marfre02-ARM) [inline tests/vkpositivelayertests.cpp:3541]: Should this be something along the lines of "Use VK_KHR_draw_indirect_count in 1.2 with feature bit enabled"

Comment (ncesario-lunarg) [reply tests/vkpositivelayertests.cpp:3541]: Yep, that makes way more sense, thanks!

**Comment by ncesario-lunarg** [comment]
It sounds like this change resolves the original issue and leaves everyone happy (more or less)?

File: layers/drawdispatch.cpp [hunk 2]
```diff
@@ -844,7 +844,8 @@ bool CoreChecks::ValidateCmdDrawIndirectCount(VkCommandBuffer commandBuffer, VkB
                                               VkBuffer countBuffer, VkDeviceSize countBufferOffset, uint32_t maxDrawCount,
                                               uint32_t stride, const char *apiName) const {
     bool skip = false;
-    if ((api_version >= VK_API_VERSION_1_2) && (enabled_features.core12.drawIndirectCount == VK_FALSE)) {
+    if ((device_extensions.vk_khr_draw_indirect_count != kEnabledByCreateinfo) &&
+        ((api_version >= VK_API_VERSION_1_2) && (enabled_features.core12.drawIndirectCount == VK_FALSE))) {
```
Comment (sfricke-samsung) [inline layers/drawdispatch.cpp:848]: As a mental sanity check (and plan to reuse later for #2619)
 
| case | 1.0 | 1.1 | 1.2 |
|--------|------|------|------|
| nothing enabled | error | error | error |
| enable ext | valid | valid | valid |
| enable drawIndirectCount | N/A | N/A | valid |
| enable both | N/A | N/A | error (VUID `02831`) |

This logic matches what I was expecting, (it adds the 4th `valid` case from throwing a false error)

Test also cover cases

:ship:

**Review by sfricke-samsung — APPROVED** [review]

**Pull request merged**